packages/salesforcedx-vscode-apex/src/codecoverage/colorizer.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { Range, TextDocument, TextEditor, window, workspace } from 'vscode';
+import {
+  coveredLinesDecorationType,
+  uncoveredLinesDecorationType
+} from './decorations';
+import { StatusBarToggle } from './statusBarToggle';
+
+const apexDirPath = path.join(
+  workspace!.workspaceFolders![0].uri.fsPath,
+  '.sfdx',
+  'tools',
+  'testresults',
+  'apex'
+);
+
+export function getLineRange(
+  document: TextDocument,
+  lineNumber: number
+): Range {
+  const adjustedLineNumber = lineNumber - 1;
+  const firstLine = document.lineAt(adjustedLineNumber);
+
+  return new Range(
+    adjustedLineNumber,
+    firstLine.range.start.character,
+    adjustedLineNumber,
+    firstLine.range.end.character
+  );
+}
+
+export type CoverageTestResult = {
+  coverage: {
+    coverage: CoverageItem[];
+  };
+};
+
+export type CoverageItem = {
+  id: string;
+  name: string;
+  totalLines: number;
+  lines: { [key: string]: number };
+};
+
+function getTestRunId() {
+  const testRunIdFile = path.join(apexDirPath, 'test-run-id.txt');
+  if (!fs.existsSync(testRunIdFile)) {
+    throw new Error('No test run information was found for this project.');
```

Comment:
Error when the project does not have any test run information, this would happen if they've never run tests via vscode/cli.

Reply:
Can we add a next step? Something like, 'No test run information was found for this project. Set "salesforcedx-vscode-core.retrieve-test-code-coverage": true in your user or workspace settings, then run Apex tests from the Apex Tests sidebar or from within a test class file.'